include/dlaf/factorization/cholesky/mc.h

```diff
@@ -126,74 +141,98 @@ void Cholesky<Backend::MC, Device::CPU, T>::call_L(comm::CommunicatorGrid grid,
-                            mat_a.tileSize(kk_idx), kk_rank.row());
+    const GlobalTileIndex kk_idx(k, k);
+    const comm::Index2D kk_rank = distr.rankGlobalTile(kk_idx);
+
+    const LocalTileSize kk_offset{
+        distr.nextLocalTileFromGlobalTile<Coord::Row>(k),
+        distr.nextLocalTileFromGlobalTile<Coord::Col>(k),
+    };
+
+    const LocalTileIndex diag_wp_idx{0, kk_offset.cols()};
+
+    auto& panel_col = panel_cols.next_resource();
+    auto& panel_col_t = panel_cols_t.next_resource();
+
+    panel_col.set_offset(kk_offset);
+    panel_col_t.set_offset(kk_offset);
+
+    // Factorization of diagonal tile and broadcast it along the `k-th column
+    if (kk_rank.col() == this_rank.col()) {
+      if (kk_rank.row() == this_rank.row()) {
+        potrf_diag_tile(executor_hp, mat_a(kk_idx));
+        if (k != nrtile - 1) {
+          panel_col_t.set_tile(diag_wp_idx, mat_a.read(kk_idx));
+          dataflow(executor_mpi, comm::sendTile_o, mpi_task_chain(), Coord::Col,
+              panel_col_t.read(diag_wp_idx));
+        }
+      }
+      else {
+        if (k != nrtile - 1)
+          dataflow(executor_mpi, comm::recvTile_o, mpi_task_chain(), Coord::Col, panel_col_t(diag_wp_idx),
+              kk_rank.row());
+      }
     }
 
-    // Iterate over the k-th column
-    for (SizeType i = k + 1; i < nrtile; ++i) {
-      GlobalTileIndex ik_idx(i, k);
-      comm::Index2D ik_rank = mat_a.rankGlobalTile(ik_idx);
+    if (k == nrtile - 1)
+      continue;
 
-      if (this_rank == ik_rank) {
-        trsm_panel_tile(executor_hp, panel[k], mat_a(ik_idx));
-        panel[i] = mat_a.read(ik_idx);
-        dataflow(executor_mpi, comm::sendTile_o, mpi_task_chain(), Coord::Row, panel[i]);
-      }
-      else if (this_rank.row() == ik_rank.row()) {
-        panel[i] = dataflow(executor_mpi, comm::recvAllocTile<T>, mpi_task_chain(), Coord::Row,
-                            mat_a.tileSize(ik_idx), ik_rank.col());
+    // COLUMN UPDATE
+    for (SizeType i = distr.nextLocalTileFromGlobalTile<Coord::Row>(k + 1);
+         i < distr.localNrTiles().rows(); ++i) {
+      const LocalTileIndex local_idx(Coord::Row, i);
+      const LocalTileIndex ik_idx(i, distr.localTileFromGlobalTile<Coord::Col>(k));
+
+      if (kk_rank.col() == this_rank.col()) {
+        trsm_panel_tile(executor_hp, panel_col_t.read(diag_wp_idx), mat_a(ik_idx));
+
+        panel_col.set_tile(local_idx, mat_a.read(ik_idx));
       }
     }
 
-    // Iterate over the trailing matrix
-    for (SizeType j = k + 1; j < nrtile; ++j) {
-      GlobalTileIndex jj_idx(j, j);
-      comm::Index2D jj_rank = mat_a.rankGlobalTile(jj_idx);
+    // row panel has been used for temporary storage of diagonal panel for column update
+    panel_col_t.reset();
 
-      if (this_rank.col() != jj_rank.col())
+    // TODO skip last step tile
+    broadcast(executor_mpi, kk_rank.col(), panel_col, panel_col_t, mpi_task_chain);
```

Comment:
This is a point we may want to discuss.

Reply:
Regarding this, we may want to open an issue about the limitation, so that we can address this in another PR and keep it as it is for the moment (at each iteration there will be a superfluous communication).